crates/biome_service/src/workspace/server.rs

```diff
@@ -573,6 +573,20 @@ impl WorkspaceServer {
                 .is_some_and(|ignored_matches| ignored_matches.is_ignored(path, is_dir(path)))
     }
 
+    pub(super) fn is_ignored_by_scanner(&self, project_key: ProjectKey, path: &Utf8Path) -> bool {
+        let Some(settings) = self.projects.get_root_settings(project_key) else {
+            return true; // If the project isn't loaded, nothing should be scanned.
```

Comment:
Shouldn't this be an error? What's a use case where it's valid to have root settings missing?